core/api.go

```diff
@@ -243,6 +243,28 @@ type page struct {
 	LastPage bool         `json:"last_page"`
 }
 
+func (a *API) authnHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		req = (&authn.API{
+			Tokens:           a.accessTokens,
+			TokenMap:         make(map[string]authn.TokenResult),
```

Comment:
It looks like this is making a new map for every request, which would defeat the purpose of putting tokens in this map as a cache. Maybe pull this object out of the closure just before line 247?

It might also be a little clearer to define a constructor in package authn:

```go
func (a *API) authnHandler(handler http.Handler) http.Handler {
	auth := authn.NewAPI(a.accessTokens, networkPrefix)
	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
		req = auth.Authenticate(req)
		handler.ServeHTTP(rw, req)
	})
}
```

Doing this might also allow us to un-export authn.TokenResult???